slick/src/main/scala/slick/jdbc/SQLiteProfile.scala

```diff
@@ -94,39 +94,48 @@ trait SQLiteProfile extends JdbcProfile {
   )
 
   class ModelBuilder(mTables: Seq[MTable], ignoreInvalidDefaults: Boolean)(implicit ec: ExecutionContext) extends JdbcModelBuilder(mTables, ignoreInvalidDefaults) {
+
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta)
     override def createPrimaryKeyBuilder(tableBuilder: TableBuilder, meta: Seq[MPrimaryKey]): PrimaryKeyBuilder = new PrimaryKeyBuilder(tableBuilder, meta)
 
     class ColumnBuilder(tableBuilder: TableBuilder, meta: MColumn) extends super.ColumnBuilder(tableBuilder, meta) {
-      /** Regex matcher to extract name and length out of a db type name with length ascription */
-      final val TypePattern = "^([A-Z\\s]+)(\\(([0-9]+)\\))?$".r
-      private val (_dbType,_size) = meta.typeName match {
-        case TypePattern(d,_,s) => (d, Option(s).map(_.toInt))
-        case "" => ("TEXT", None)
-      }
+
+      final val TypePattern = "^([A-Z\\s]+)(\\(\\s?([0-9]+)\\s?,?\\s?([0-9]+)?\\s?\\))?$".r
+
+      /** Regex matcher to extract name and length
+       * out of a db type name with length ascription
+       */
+      def extractTypeProps(typeName: String = "NUMERIC(18,2)"): ( String, Option[ Int ] ) =
+        typeName match {
+          case TypePattern(name, _, precision, _) => ( name, Option( precision ).map( _.toInt ) )
+          case "" => ( "TEXT", None )
+        }
+
+      private val ( _dbType, _size ) = extractTypeProps(meta.typeName)
```

Comment:
Using the underscore at the start of a name in Scala feels alien to me. Could it be without?

Reply:
@hvesalai, the code was already like that before this PR was made and I didn't want to introduce even more changes. 

It also disturbs me as many other poorly formatted pieces in the whole code base. I would prefer to tackle that in another PR. Maybe we should add scalafmt and have a clean-up hackathon.

Reply:
Ok